src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -259,45 +333,46 @@ object DataModelInstanceRelation {
         case null if !nullable =>
           throw new CdfSparkException(propertyNotNullableMessage(propertyType)) // scalastyle:off null
         case null => Json.Null // scalastyle:off null
-        case x: Iterable[String] @unchecked if x.nonEmpty && x.head.isInstanceOf[String] =>
+        case x: Iterable[_] if x.isEmpty => Json.arr()
+        case x: Iterable[String] @unchecked if x.head.isInstanceOf[String] =>
           Json.fromValues(x.map(Json.fromString))
```

Comment:
I think it would be nice to add a default case to all these `match` and throw an explicit error. just something like `throw Exception(s"$x is not valid text[]")`. In case we missed something here, it'd be easier to figure out what's wrong

Reply:
done